### PR TITLE
Evolute the reset feature gate to Beta

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Evolute the reset feature gate to Beta.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
None

### CODE REVIEW

File: docs/userguide/failover/failover-overview.md [hunk 1]
```diff
@@ -25,15 +25,15 @@ When cluster `member1` fails, pod instances on the cluster are evicted and migra
 
 ## How Do I Enable the Feature?
 
-The failover feature is controlled by the `Failover` feature gate, users need to enable the `Failover` feature gate of `karmada-controller`:
+The failover feature is controlled by the `Failover` feature gate. `Failover` feature gate has evolved to Beta since Karmada v1.4. If users use Karmada 1.3 or earlier, they need to enable the `Failover` feature gate of `karmada-controller`:
```
Comment (jwcesign) [inline docs/userguide/failover/failover-overview.md:28]: this feature is enabled by default from v1.4.0, it needs to described here.

**Comment by jwcesign** [comment]
cc @RainbowMango for the final check

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**